---
title: "Kalahari Fungi - Analysis in R to follow the upstream preparation in file kf_analysis.sh and QIIME-notes.txt"
output:
  html_document:
    toc: yes
---
```{r set-options, echo=FALSE, cache=FALSE}
options(width = 400)
# create folders for output of tables and figures
if (!dir.exists("tables")) {dir.create("tables")}
if (!dir.exists("figures")) {dir.create("figures")}
```

Analysis of fungal sequence data derived from the same DNA extractions as those used in our previous bacterial analysis ([Elliott et al 2014](http://dx.doi.org/10.1007/s10531-014-0684-8)). The analysis for that paper can be found [here](http://davidelliott.github.io/kalahari_bacteria/kalahari_BSC_bacteria.html). 

# Revision notes
This analysis was updated after reviewer feedback as follows:


* Use latest version of UNITE for taxonomy (upstream from this script, documented in QIIME-notes.txt)
* Due to better taxonomy in the new UNITE, the custom approach to taxonomy is removed
* Slightly alter the DESeq analysis to work on the exact same subsetted objects as was used for the ADONIS - this makes a bit more logical sense and does not affect the results much. Specifically, the depth analysis is now done only on the open grass sites (=cyanobacterial biocrust areas), and not in the canopy places.
* Use volcano plots to display the DESeq2 results. This includes using a more sophisticated approach to OTU selection for comparisons

# Import data and setup

## Fungi
```{r, import_fungi, echo=FALSE,message=FALSE, warning=FALSE}
library("phyloseq"); library("ggplot2"); library("scales"); library("grid"); library("reshape2"); library(RColorBrewer); library("DESeq2"); library("xtable"); library("vegan"); library("ape"); library(qiime2R); library(EnhancedVolcano);library(ggpubr);library(gridExtra)

# BiocManager::install('EnhancedVolcano')

biom <- import_biom("data/fungi/otutab.biom",taxaPrefix="X")
map <- import_qiime_sample_data("data/fungi/map.txt")

map$zone2line <- sub(pattern="_",replacement="\n",x=map$zone)

expt <- qza_to_phyloseq(features = "data/fungi/feature-table-1.qza",
                taxonomy = "data/fungi/unite-v9-dynamic-vsearch-classifications.qza")

expt <- merge_phyloseq(expt, map)


# add bacterial phylum to the environmental data
# bac_phylum <- read.csv("data/otu_table_bacteria_phylum_rel.csv",na.strings = "")
#bac_phylumt = setNames(data.frame(t(bac_phylum[,-1])), bac_phylum[,1])
#colnames(bac_phylumt) <- gsub(pattern = "-",replacement = ".",x = colnames(bac_phylumt))
#sample_data(expt) <- cbind(sample_data(expt), bac_phylumt[row.names(map),])

tidy_phyloseq <- function(my_phyloseq){
  # Fix ranks. 
  colnames(tax_table(my_phyloseq)) = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
  # fix taxa names
  tax_table(my_phyloseq)[,colnames(tax_table(my_phyloseq))] <- gsub(tax_table(my_phyloseq)[,colnames(tax_table(my_phyloseq))],pattern="[a-z]__",replacement="")
  return(my_phyloseq)
}

# from_phyloseq_web 
veganotu = function(physeq) {
  require(vegan)
    OTU = otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU = t(OTU)
    }
    return(as(OTU, "matrix"))
}

expt <- tidy_phyloseq(expt)

# Add new names to sample data for plots
sample_data(expt)$depth2 <- "0-1 cm"
sample_data(expt)$depth2[sample_data(expt)$depth == 1] <- "1-2 cm"

samples <- as.character(sample_data(expt)$X.SampleID)
sample_data(expt)$samples <- as.factor(substr(samples, 1, nchar(samples)-1))

sample_data(expt)$zc_depth <- paste(sample_data(expt)$zone_code,sample_data(expt)$depth,sep="")

expt.rel <- transform_sample_counts(expt, function(x) 100*x/(sum(x)))
# check that the above worked properly:
# sample_sums(expt.rel)
# yes - all samples add up to 100

print(expt)
map.df <- as.data.frame(sample_data(expt))

map.df$n.sequences <- colSums(otu_table(expt))[row.names(map.df)]

expt.rel.class <- tax_glom(expt.rel,taxrank="Class")
rank_names(expt)

# write.csv(otu_table(expt),"tables/17_otu_count.csv")
# write.csv(otu_table(expt.rel),"tables/17_otu.csv")

# Prepare some phyloseq objects for later use
expt.rel.phylum <- tax_glom(expt.rel,taxrank="Phylum")

expt.rel.phylum.all <- merge_samples(expt.rel.phylum,"ReversePrimer")
expt.rel.phylum.all <- transform_sample_counts(expt.rel.phylum.all, function(x) x/48)
expt.rel.phylum.type <- merge_samples(expt.rel.phylum,"type")
expt.rel.phylum.type <- transform_sample_counts(expt.rel.phylum.type, function(x) x/24)
expt.rel.phylum.veg <- merge_samples(expt.rel.phylum,"zone")
expt.rel.phylum.veg <- transform_sample_counts(expt.rel.phylum.veg, function(x) x/12)

taxa_names(expt.rel.phylum.all) <- tax_table(expt.rel.phylum.all)[,2]
sample_names(expt.rel.phylum.all) <- "overall"
taxa_names(expt.rel.phylum.type) <- tax_table(expt.rel.phylum.type)[,2]
taxa_names(expt.rel.phylum.veg) <- tax_table(expt.rel.phylum.veg)[,2]
```


## Bacteria
Bacterial data is from Elliott et al (2014)

```{r bac_import, echo=FALSE,message=FALSE}

# set input filenames
otufilename <- "data/bacteria/otu_table.biom"
mapfilename <- "data/bacteria/Elliott_2188B_map_sff.txt"

biom <- import_biom(otufilename,taxaPrefix="X")
map <- import_qiime_sample_data(mapfilename=mapfilename)
bacteria <- merge_phyloseq(biom, map)
# Fix ranks. 
colnames(tax_table(bacteria)) = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
# fix taxa names
tax_table(bacteria)[,colnames(tax_table(bacteria))] <- gsub(tax_table(bacteria)[,colnames(tax_table(bacteria))],pattern="[a-z]__",replacement="")

# Add new names to sample data for plots
sample_data(bacteria)$depth2 <- "0-1 cm"
sample_data(bacteria)$depth2[sample_data(bacteria)$depth == 1] <- "1-2 cm"

# Remove Mitochondria and Chloroplast. No mitochondria found.
bacteria <- subset_taxa(bacteria,Class!="Chloroplast")

print(bacteria)
```


```{r set_formatting,echo=FALSE}
# set default ggplot theme
theme_set(theme_bw(base_size = 12, base_family = ""))
alpha = 0.75
# define colour scales
#color_var = c(levels((get_variable(expt, "type"))), levels((get_variable(expt, "zone_code"))), levels((get_variable(expt, "cover"))))
color_var = c(levels((get_variable(expt, "type"))), levels((get_variable(expt, "zone_code"))))

color_pal = brewer.pal(length(color_var), "Set1")
names(color_pal) <- color_var
#length(color_pal)
color_pal["T"] <- "#F781BF" # don't like yellow
color_pal["0-1 cm"] <- color_pal["crust"]
color_pal["1-2 cm"] <- color_pal["subsoil"]

fill_pal <- color_pal
fill_pal["subsoil"] <- "white"
fill_pal["open"] <- "white"
fill_pal["1-2 cm"] <- fill_pal["subsoil"]
fill_pal["0-1 cm"] <- fill_pal["crust"]

depth_fill <- color_pal[c(7,8)]

# define shape scale
shape_var = c(levels((get_variable(expt, "type"))), "0-1 cm", "1-2 cm")
shape_scale = c(19,24,19,24)
names(shape_scale) = c(shape_var)

# copy parts of the colour scales for convenience:
zone_colours <- color_pal[c("AG",      "PG",      "S",       "T")]

# zone/depth shapes and colours - added during proofing stage

zcd <- sort(unique(map.df$zc_depth))

zcd_shapes <- c(21,21,22,22,23,23,24,24)
zcd_colours <- color_pal[c("AG","AG","PG","PG","S","S","T","T")]
zcd_fill <- zcd_colours
zcd_fill[c(2,4,6,8)] <- "#FFFFFF"
zcd_solid <- c(TRUE,FALSE,TRUE,FALSE,TRUE,FALSE,TRUE,FALSE)

names(zcd_shapes) = zcd
names(zcd_colours) = zcd
names(zcd_fill) = zcd
names(zcd_solid) = zcd

zone_shapes <- zcd_shapes[c(1,3,5,7)]
names(zone_shapes) <- names(zone_colours) 

```

# Main Tables & Figures
## Figure 1. Schematic representation of hypothetical roles for fungi in biocrusts, and expected distribution consistent with these roles. 
(not shown here)

## Figure 2. Mixed tree, shrub and grass rangeland at the study site. 
Extensive bare soil patches as shown here between grasses are common and support the development of biocrusts. Photographs of individual vegetation types can be found in Elliott et al (2014). (not shown here)

## Figure 3. Fungal class abundance by vegetation zone and depth in Kalahari arenosols, southern Botswana. 

```{r class_plots_alt2,echo=FALSE,message=FALSE, fig.width=8, fig.height=4, warning=FALSE}

TopNOTUs <- names(sort(taxa_sums(expt.rel.class), TRUE)[1:10])
expt.rel.class.topN <- prune_taxa(TopNOTUs, expt.rel.class)
c <- tax_table(expt.rel.class.topN)[,"Class"]
p <- tax_table(expt.rel.class.topN)[,"Phylum"]
taxa_names(expt.rel.class.topN) <- paste(p,"__",c,sep="")
my_phyloseq <- expt.rel.class.topN

# get the OTU tables
  OTUs.table <- data.frame(otu_table(my_phyloseq))
  OTUs.t <- t(OTUs.table)
  
  sample_data(my_phyloseq)$zone_type <- paste(sample_data(my_phyloseq)$zone_code,sample_data(my_phyloseq)$type,sep="_")
  
  sample_data(my_phyloseq)$sample <- row.names(sample_data(my_phyloseq))
  sample_data <- sample_data(my_phyloseq)[,c("sample","zone","zone_code","zone2line","cover","type","depth2","month")]

  # combine OTU and sample data 
  OTUs_and_env_data <- data.frame(sample_data,OTUs.t)
  
  OTUs_and_env_data.melt <- melt(OTUs_and_env_data)
 # re-name the OTU's that had names changed by melting
  OTUs_and_env_data.melt$variable <- gsub(pattern="X",replacement="",x=OTUs_and_env_data.melt$variable)
# Add linebreak in the OTU name (the OTUs were strategically named a few lines up)
OTUs_and_env_data.melt$variable <- gsub(pattern="__",replacement="\n",x=OTUs_and_env_data.melt$variable)
OTUs_and_env_data.melt$variable <- gsub(pattern="_cls_Incertae_sedis",replacement="",x=OTUs_and_env_data.melt$variable)

  # plot the OTUs

    zone_plot2 <- (ggplot(data=OTUs_and_env_data.melt,aes(x=zone_code,y=value, fill=depth2))
                  + facet_wrap(~variable,nrow = 2) 
                  + theme_bw(base_size = 9, base_family = "") 
                  + theme(aspect.ratio=1)
                  + scale_x_discrete(name="nearby vegetation") 
                  + scale_y_continuous(name="% abundance",trans = 'log10',breaks=c(0.1,1,10,100))  
                  + geom_boxplot()  
                  + theme(axis.text.x = element_text(angle = -90, vjust = 0.4))
                  + scale_fill_manual(values=fill_pal,name="Depth") )

    
```

Boxes represent the interquartile range (IQR), and error bars extend to the most extreme values within 1.5 * IQR of the box. Median values are shown as a line within the box and outliers are shown as black spots. Sample coding: AG=annual grass, PG=perennial grass, S=shrub, T=tree. The ten most abundant classes are shown, accounting for `r sum(taxa_sums(expt.rel.class.topN)/nsamples(expt.rel.class.topN)) ` of sequence reads.
```{r class_plots_alt_show,echo=FALSE,message=FALSE, fig.width=8, fig.height=4, warning=FALSE}
zone_plot2
```

## Figure 4. Correspondence analysis of the fungal community in Kalahari arenosols, southern Botswana constrained by vegetation and depth. 
Markers indicate the fungal communities of individual samples. Filled markers indicate soil surface/biocrust (0-1 cm depth), open markers indicate subsoil communities (1-2 cm depth). Sample coding: AG=annual grass, PG=perennial grass, S=shrub, T=tree.

```{r ord, echo=FALSE}
  expt.ord <- expt

ord <- ordinate(expt.ord,method = "CCA",distance = "bray",formula = ~zone+type)
p.ord <-   plot_ordination(expt.ord, ord, type = "sites", color = "zone_code", shape="zone_code") + geom_point(alpha=0.7,aes(fill=zc_depth)) + theme(aspect.ratio=1) #+ scale_colour_manual(values=zone_colours, name="Vegetation") + 
  #scale_fill_manual(values=zcd_fill, name="Vegetation", guide="none") + 
 # scale_shape_manual(values=zone_shapes, name="Vegetation") 

p.ord$layers <- p.ord$layers[-1]
p.ord

```


## checking assumptions for ADONIS2 permanova test:

```{r adnois2_setup,message=FALSE,echo=TRUE,warning=FALSE,cache=TRUE}
  # ADONIS to be done separately for open and canopy areas, at OTU/species rank.
  
  # grass areas
  expt.open <- subset_samples(expt, cover=="open")
  dist.open = phyloseq::distance(expt.open, "bray")
  
  # canopy areas
  expt.canopy <- subset_samples(expt, cover=="canopy")
  dist.canopy = phyloseq::distance(expt.canopy, "bray")
 
  # Checking for homogeneity of group dispersions to ensure that dispersion within groups does not skew the PERMANOVA test.

  ## Calculate multivariate dispersions
  mod.open.zone <- betadisper(dist.open,group = sample_data(expt.open)$zone) # need to check the ordering of data matches up - done - it does match
  mod.open.month <- betadisper(dist.open,group = sample_data(expt.open)$month)
  mod.open.type <- betadisper(dist.open,group = sample_data(expt.open)$type)
  mod.canopy.zone <- betadisper(dist.canopy,group = sample_data(expt.canopy)$zone)
  mod.canopy.month <- betadisper(dist.canopy,group = sample_data(expt.canopy)$month)
  mod.canopy.type <- betadisper(dist.canopy,group = sample_data(expt.canopy)$type)
  
  ## Perform tests
  anova(mod.open.zone)
  anova(mod.open.month)
  anova(mod.open.type)
  anova(mod.canopy.zone)
  anova(mod.canopy.month)
  anova(mod.canopy.type)
  
  # The plots and further analyses below were used to check further
  # but are commented out here to reduce the amount of outputs.
  
  ## Plot the groups and distances to centroids on the
  ## first two PCoA axes
plot(mod.open.zone)
plot(mod.open.month)
plot(mod.open.type)
plot(mod.canopy.zone)
plot(mod.canopy.month)
plot(mod.canopy.type)
  
  ## Draw a boxplot of the distances to centroid for each group
boxplot(mod.open.zone)
boxplot(mod.open.month)
boxplot(mod.open.type)
boxplot(mod.canopy.zone)
boxplot(mod.canopy.month)
boxplot(mod.canopy.type)
  
  ## Permutation test for F
permutest(mod.open.zone, pairwise = TRUE, permutations = 99)
permutest(mod.open.month, pairwise = TRUE, permutations = 99)
permutest(mod.open.type, pairwise = TRUE, permutations = 99)
permutest(mod.canopy.zone, pairwise = TRUE, permutations = 99)
permutest(mod.canopy.month, pairwise = TRUE, permutations = 99)
permutest(mod.canopy.type, pairwise = TRUE, permutations = 99)

```


## Table 1. Permutational analysis of variance to identify significant differences in fungal community structure with respect to vegetation zone, depth, and season in Kalahari arenosols of southern Botswana. 
This table shows results for the community data at OTU (97 % similarity). The “all samples” analysis was also carried out at multiple taxonomic ranks with similar results as shown in supplementary data. 

(table needs to be made manually from the results below)

```{r adonis2,message=FALSE,echo=FALSE,warning=FALSE,cache=TRUE,}

  NMDS = ordinate(expt.open, "NMDS", dist.open)
  print(adonis2(dist.open ~ zone + depth + month , as(sample_data(expt.open), "data.frame")) )

  NMDS = ordinate(expt.canopy, "NMDS", dist.canopy)
  print(adonis2(dist.canopy ~ zone + depth + month , as(sample_data(expt.canopy), "data.frame")) )


```

(see also Note S1)


```{r deseq_depth, fig.width=12,fig.height=10,echo=FALSE, message=FALSE, warning=FALSE}

  # using the same subsetted object as was used in ADONIS - grass sites only because the ADONIS indicated a depth related difference only on the grass (open) sites.
  expt.deseq <- expt.open 

  expt.deseq <- tidy_phyloseq(expt.deseq)
  # prune OTUs that are not present in at least one sample
  expt.deseq = prune_taxa(taxa_sums(expt.deseq) > 0, expt.deseq)
  
  # add 1 to the whole OTU table, to avoid infinity problems
  # this will reduce the fold changes
  otu_table(expt.deseq) <- otu_table(expt.deseq)+1
  
  typedds = phyloseq_to_deseq2(expt.deseq, ~ type)
  typedds = DESeq(typedds, test="Wald",fitType="parametric")
  # DESeq docs:
  #On p-values:
  # By default, independent filtering is performed to select a set of genes for multiple test correction which will optimize the number of adjusted p-values less than a given critical value alpha (by default 0.1). The adjusted p-values for the genes which do not pass the filter threshold are set to NA. 
  
# When using local regression fit, the user should examine plotDispEsts(dds)
# to make sure the fitted line is not sharply curving up or down based on
# the position of individual points.

  #  plotDispEsts(typedds,main="model check")

  res = results(typedds, cooksCutoff = FALSE, alpha = 0.05)
  alpha = 0.05
  sigtab_type = res
  
  # add observed data to deseq results

  expt.rel.open <- subset_samples(expt.rel, cover=="open")
# Division by 12 is because phyloseq "sums" in merge_samples, and there are 12 samples of each
  otus.rel.open.by.type <- otu_table(merge_samples(expt.rel.open,group = "type"))/12
  orbt <- as.data.frame(t(otus.rel.open.by.type))

#length(row.names(sigtab_type))
#length(row.names(orbt))

sigtab_type <- cbind(sigtab_type[row.names(sigtab_type),],orbt[row.names(sigtab_type),])
# add a difference column
sigtab_type$diff.depth <- sigtab_type$crust - sigtab_type$subsoil
# Reverse the sign of the fold change so that it represents crust:subsoil not subsoil:crust
sigtab_type$log2FoldChange <- sigtab_type$log2FoldChange * -1
# Sanity check on the model vs actual observed difference
sigtab_type$sane.depth <- sign(sigtab_type$log2FoldChange)==(sign(sigtab_type$diff.depth))
# add a link to a figure that will be made later
sigtab_type$fig <- paste("<a href='figures_extra/sig_type_",row.names(sigtab_type),".png'>plot</a>",sep="")
# store the results object for creating a plot
deseq_depth_results <- res
```

```{r deseq_cover, fig.width=12,fig.height=10,echo=FALSE, message=FALSE, warning=FALSE}

# re-written, to be removed

  # POSSIBLE ERROR. Do not appear to have subset the expt object here - need to check if that was intended. 
  # Have checked and that was the intention so there is no error, this is as stated in the methods. I.E. the "depth" analysis is done on all samples.
  # It is intended to look into the distribution of fungi in relation to the presence of grasses vs trees/shrubs
  # As done here, it does include the sub-surface soil in addition to the biocrust. That is consistent with the hypotheses, it would also be valid to limit it to the biocrust.

  expt.deseq <- expt

  # add 1 to the whole OTU table, to avoid infinity problems
  # this will reduce the fold changes
  otu_table(expt.deseq) <- otu_table(expt.deseq)+1
  
  coverdds = phyloseq_to_deseq2(expt.deseq, ~ cover)
  coverdds = DESeq(coverdds, test="Wald",fitType="parametric")

  # plotDispEsts(coverdds,main="model check")

  res = results(coverdds, cooksCutoff = FALSE, alpha = 0.05)
  alpha = 0.05
  sigtab_cover = res
  
  # add observed data to deseq results
# Division by 24 is because phyloseq "sums" in merge_samples, and there are 24 samples of each
  otus.rel.by.cover <- otu_table(merge_samples(expt.rel,group = "cover"))/24
  orbc <- as.data.frame(t(otus.rel.by.cover))

  #sigtab_cover <- cbind(sigtab_cover,orbc)
  sigtab_cover <- cbind(sigtab_cover[row.names(sigtab_cover),],orbc[row.names(sigtab_cover),])
  
  # add a difference column

  sigtab_cover$diff.cover <- sigtab_cover$open - sigtab_cover$canopy
# Sanity check on the model vs actual observed difference
sigtab_cover$sane.cover <- sign(sigtab_cover$log2FoldChange)==(sign(sigtab_cover$diff.cover))
# add a link to a figure that will be made later
sigtab_cover$fig <- paste("<a href='figures_extra/sig_cover_",row.names(sigtab_cover),".png'>plot</a>",sep="")
# store the results object for creating a plot
deseq_cover_results <- res
```

Use differential abundance analysis to identify which fungi are associated with the cyanobacterial biocrust containing areas (open grass interspaces)
```{r DESeq_new1, echo=FALSE, message=FALSE, warning=FALSE}
# Identify fungi which are associated with the biocrust areas (open grass interspaces)

# set a variable for the desired log change cut-off
log_threshold <- 2 # set at 4-fold

  # use all the full experiment data in this first test
  expt.deseq <- expt

  # add 1 to the whole OTU table, to avoid infinity problems
  # this will reduce the fold changes slightly
  otu_table(expt.deseq) <- otu_table(expt.deseq)+1
  
  coverdds = phyloseq_to_deseq2(expt.deseq, ~ cover)
  coverdds = DESeq(coverdds, test="Wald",fitType="parametric")

  # plotDispEsts(coverdds,main="model check")

  res = results(coverdds, cooksCutoff = FALSE, alpha = 0.05)
  alpha = 0.05
  deseq_cover_results <- res
  
# From the initial DESeq analysis on all samples with respect to cover type
# identify the OTUs which are associated with the open areas (grass interspaces) which harbour cyanobacterial biocrusts
# these will be the ones where Log change is >threshold and P<0.05

index1 <- deseq_cover_results$log2FoldChange>log_threshold # grass interspaces
index2 <- deseq_cover_results$log2FoldChange<(log_threshold*-1) # tree/shrub canopy
index3 <- deseq_cover_results$padj<0.05

cat("grass interspace associated OTUs (place where cyanobacterial biocrusts are)")
(biocrust_associated_OTUs <- row.names(deseq_cover_results)[index1&index3])
cat("canopy associated OTUs (place lacking cyanobacterial biocrusts)")
(canopy_associated_OTUs <- row.names(deseq_cover_results)[index2&index3])

```

Check to see if any of the biocrust associated fungi preferentially occupy the surface or sub-surface soil. Also perform the same check on the fingi which are associated with the non-biocrust areas (canopy).
```{r DESeq_new2, echo=FALSE, message=FALSE, warning=FALSE}

# Now perform the differential abundance analysis on the biocrust associated OTUs to see if any of them preferentially occupy the surface or sub-surface soil

expt.biocrust <- prune_taxa(biocrust_associated_OTUs , expt)

  expt.deseq <- expt.biocrust
  # add 1 to the whole OTU table, to avoid infinity problems
  # this will reduce the fold changes slightly
  otu_table(expt.deseq) <- otu_table(expt.deseq)+1
  
  bc_depth_dds = phyloseq_to_deseq2(expt.deseq, ~ depth)
  bc_depth_dds = DESeq(bc_depth_dds, test="Wald",fitType="parametric")

  # plotDispEsts(bc_depth_dds,main="model check")

  res = results(bc_depth_dds, cooksCutoff = FALSE, alpha = 0.05)
  alpha = 0.05

  deseq_biocrust_by_depth_results <- res

# same again for canopy assocaited taxa
  
expt.canopy <- prune_taxa(canopy_associated_OTUs , expt)

  expt.deseq <- expt.canopy

  # add 1 to the whole OTU table, to avoid infinity problems
  # this will reduce the fold changes slightly
  otu_table(expt.deseq) <- otu_table(expt.deseq)+1
  
  depthdds = phyloseq_to_deseq2(expt.deseq, ~ depth)
  depthdds = DESeq(depthdds, test="Wald",fitType="parametric")

  # plotDispEsts(coverdds,main="model check")

  res.depth = results(depthdds, cooksCutoff = FALSE, alpha = 0.05)
  alpha = 0.05
  # sigtab_depth = res.depth
  
  deseq_canopy_by_depth_results <- res.depth

# for the depth results we are interested in both the OTUs that have significantly differential abundance AND those that do not
# To highlight this in figure, we can label those OTUS - which means most of them, but excluding the ambiguous few in the middle
# Criteria:
  # P<0.05 & fold change >log_threshold
  # P>0.05 & fold change <log_threshold
  subsurface_OTUs.open <- deseq_biocrust_by_depth_results$padj<0.05 & abs(deseq_biocrust_by_depth_results$log2FoldChange)>log_threshold
  cosmopolitan_OTUs.open <- deseq_biocrust_by_depth_results$padj>0.05 & abs(deseq_biocrust_by_depth_results$log2FoldChange)<log_threshold

  wanted_labels_index.open <- subsurface_OTUs.open | cosmopolitan_OTUs.open
  
  subsurface_OTUs.canopy <- deseq_canopy_by_depth_results$padj<0.05 & abs(deseq_canopy_by_depth_results$log2FoldChange)>log_threshold
  cosmopolitan_OTUs.canopy <- deseq_canopy_by_depth_results$padj>0.05 & abs(deseq_canopy_by_depth_results$log2FoldChange)<log_threshold

  wanted_labels_index.canopy <- subsurface_OTUs.canopy | cosmopolitan_OTUs.canopy
```

Create volcano plots to graphically show the results of the differential abundance analyses

```{r volcano_plots_new, echo=FALSE, message=FALSE, warning=FALSE}
# as suggested by reviewer
# using guidance from https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html
# plotting the results of deseq_cover_results + deseq_biocrust_by_depth_results

annotation_open <- data.frame(
   x = c(-4,0,4),
   y = c(15,15,15),
   label = c("More abundant\n0-1 cm depth","Similarly abundant\nthroughout depths", "More abundant\n1-2 cm depth")
)

annotation_cover <- data.frame(
   x = c(-4.2,4.2),
   y = c(45,45),
   label = c("More abundant in canopy areas\n(these compared by\ndepth in fig 6B)", "More abundant in grass inter-spaces\n(these compared by\ndepth in fig 6A)")
)

annotation_canopy<- data.frame(
   x = c(-4,0,4),
   y = c(12,12,12),
   label = c("More abundant\n0-1 cm depth", "Similarly abundant\nthroughout depths","More abundant\n1-2 cm depth")
)

## Cover results - this plot determines which OTUs are associated to biocrust
OTU_IDs <- rownames(deseq_cover_results)
TAX_abbreviation <- as.character(substr(x = tax_table(expt)[rownames(deseq_cover_results),"Class"] , start = 0, stop = 3))
OTU_TAX <- paste(OTU_IDs,TAX_abbreviation)
OTU_TAX_labels.cover <- substr(x = OTU_TAX , start = 5, stop = 12)

volcano_cover <- EnhancedVolcano(deseq_cover_results,
    lab = OTU_TAX_labels.cover,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 0.05,
    FCcutoff = log_threshold,
    #title = 'B. Differential abundance of fungal OTUs by cover type',
    #subtitle = '(areas of grass inter-spaces vs areas under shrubs and trees)',
    pointSize = 3.5,
    labSize = 3.5,
    legendLabels=c('Not significant','Log (base 2) FC','p<0.05',
      'p<0.05 & Log(2) change>2'),
    drawConnectors = TRUE,
    arrowheads = FALSE,
    max.overlaps = 20,
    widthConnectors = 0) + geom_text(data=annotation_cover, aes( x=x, y=y, label=label), size=4 , fontface="italic") 


## depth results. This plot shows which of the biocrust associated taxa are differentially abundant by depth
# create custom names for OTU labelling

OTU_IDs <- rownames(deseq_biocrust_by_depth_results)
TAX_abbreviation <- as.character(substr(x = tax_table(expt)[rownames(deseq_biocrust_by_depth_results),"Class"] , start = 0, stop = 3))
OTU_TAX <- paste(OTU_IDs,TAX_abbreviation)
OTU_TAX_labels.depth <- substr(x = OTU_TAX , start = 5, stop = 12)


volcano_depth <- EnhancedVolcano(deseq_biocrust_by_depth_results,
    lab = OTU_TAX_labels.depth,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 0.05,
    FCcutoff = log_threshold,
   # title = 'A. Differential abundance of fungal OTUs by depth',
   # subtitle = '(areas of cyanobacterial biocrust of grass inter-spaces)',
    pointSize = 3.5,
    labSize = 3.5,
   selectLab = OTU_TAX_labels.depth[wanted_labels_index.open],
    legendLabels=c('Not significant','Log (base 2) FC','p<0.05',
      'p<0.05 & Log(2) change>2'),
    drawConnectors = TRUE,
    arrowheads = FALSE,
    max.overlaps = 20,
    widthConnectors = 0) + geom_text(data=annotation_open, aes( x=x, y=y, label=label), size=4 , fontface="italic") 


## depth results. This plot shows which of the biocrust associated taxa are differentially abundant by depth
# create custom names for OTU labelling

OTU_IDs <- rownames(deseq_canopy_by_depth_results)
TAX_abbreviation <- as.character(substr(x = tax_table(expt)[rownames(deseq_canopy_by_depth_results),"Class"] , start = 0, stop = 3))
OTU_TAX <- paste(OTU_IDs,TAX_abbreviation)
OTU_TAX_labels.depth <- substr(x = OTU_TAX , start = 5, stop = 12)


volcano_depth_canopy <- EnhancedVolcano(deseq_canopy_by_depth_results,
    lab = OTU_TAX_labels.depth,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 0.05,
    FCcutoff = log_threshold,
   # title = 'A. Differential abundance of fungal OTUs by depth',
   # subtitle = '(areas of cyanobacterial biocrust of grass inter-spaces)',
    pointSize = 3.5,
    labSize = 3.5,
   selectLab = OTU_TAX_labels.depth[wanted_labels_index.canopy],
    legendLabels=c('Not significant','Log (base 2) FC','p<0.05',
      'p<0.05 & Log(2) change>2'),
    drawConnectors = TRUE,
    arrowheads = FALSE,
    max.overlaps = 20,
    widthConnectors = 0) + geom_text(data=annotation_canopy, aes( x=x, y=y, label=label), size=4 , fontface="italic") 


# create custom names for OTU labelling

deseq_cover_plot <- volcano_cover + theme(legend.position="none") + labs(title = NULL, subtitle = NULL,caption = NULL) 
deseq_biocrust_depth_plot <- volcano_depth + theme(legend.position="none") + labs(title = NULL, subtitle = NULL,caption = NULL) + labs(tag = "A.") 
deseq_canopy_depth_plot <- volcano_depth_canopy + theme(legend.position="none") + labs(title = NULL, subtitle = NULL,caption = NULL) + labs(tag = "B.") 
deseq_volcano_legend <- as_ggplot(get_legend(volcano_depth))

# cover plot
deseq_cover_plot + labs(title="all taxa by cover type")
# depth plots
deseq_biocrust_depth_plot + labs(title="biocrust associated taxa by depth")
deseq_canopy_depth_plot + labs(title="canopy associated taxa by depth")

# Output to PDF also
pdf(file = "figures/volcano-plot-cover.pdf", width=9, height=6)
grid.newpage()
pushViewport(viewport(layout = grid.layout(nrow = 2, ncol = 1, heights = c(6,1)))) 
print(deseq_cover_plot, vp = viewport(layout.pos.row = 1))
print(deseq_volcano_legend, vp = viewport(layout.pos.row = 2))
dev.off()

pdf(file = "figures/volcano-plots-depth.pdf", width=7, height=9)
grid.newpage()
pushViewport(viewport(layout = grid.layout(nrow = 3, ncol = 1, heights = c(4,4,1)))) 
print(deseq_biocrust_depth_plot, vp = viewport(layout.pos.row = 1))
print(deseq_canopy_depth_plot, vp = viewport(layout.pos.row = 2))
print(deseq_volcano_legend, vp = viewport(layout.pos.row = 3))
dev.off()
```



```{r deseq_tables,echo=FALSE}
# the deseq analyses produced 2 tables: sigtab_cover and sigtab_type
# here they are adjusted and combined for eventual output

#### First build a master table with all the OTU information known

  bigtab = as.data.frame(tax_table(expt))

# merge the 2 deseq tables and suffix with .depth or .cover

sigtabs <- merge(as.data.frame(sigtab_cover),as.data.frame(sigtab_type),by.x=0,by.y=0,all.x=T,suffixes=c(".cover",".depth")) 

# merge the taxonomy with the deseq results
bigtab <- merge(bigtab,sigtabs,by.x=0,by.y="Row.names",all.x=T) 

# set row names as OTU names
row.names(bigtab) <- bigtab$Row.names

# prepare "subset" tables with the most significant results which will be main tables in the manuscript
wanted_cols_type <- c("Phylum",  "Class",  "Order",	"Family",	"padj.depth","crust","subsoil","diff.depth","assoc.cover")
wanted_cols_cover <- c("Phylum",  "Class",  "Order",	"Family",	"padj.cover","canopy","open","diff.cover","assoc.depth")
wanted_rows_type <- (bigtab$padj.depth<0.05) & (!is.na(bigtab$padj.depth)) & bigtab$sane.depth
wanted_rows_cover <- (bigtab$padj.cover<0.05) & (!is.na(bigtab$padj.cover)) & bigtab$sane.cover

# record in bigtab which OTUs are associated by depth or canopy type
bigtab$assoc.type <- NA
bigtab[wanted_rows_type&bigtab$diff.depth>0,"assoc.depth"] <- "0-1 cm"
bigtab[wanted_rows_type&bigtab$diff.depth<0,"assoc.depth"] <- "1-2 cm"

bigtab$assoc.cover <- NA
bigtab$assoc.cover.marker <- 3
bigtab[wanted_rows_cover&bigtab$diff.cover>0,"assoc.cover"] <- "open"
bigtab[wanted_rows_cover&bigtab$diff.cover>0,"assoc.cover.marker"] <- 64
bigtab[wanted_rows_cover&bigtab$diff.cover<0,"assoc.cover"] <- "canopy"
bigtab[wanted_rows_cover&bigtab$diff.cover<0,"assoc.cover.marker"] <- 17

sigtab_type.subset <- bigtab[wanted_rows_type,wanted_cols_type]
sigtab_cover.subset <- bigtab[wanted_rows_cover,wanted_cols_cover]
# order by diff
sigtab_type.subset <- sigtab_type.subset[order(sigtab_type.subset$diff.depth,decreasing = T),] 
sigtab_cover.subset <- sigtab_cover.subset[order(sigtab_cover.subset$diff.cover,decreasing = T),] 

```

## Table. Fungi that are significantly (adjusted p <0.05) more abundant in biocrusts (0-1 cm depth) or subsoil (1-2 cm depth) in grass interspaces on Kalahari arenosols of southern Botswana. This table is removed from the revised manuscript, replaced by a volcano plot. However, it is kept here for reference. 

```{r results='asis',echo=FALSE}
# display an html table of significant OTUs identified by DESEQ

print(type="html",xtable(sigtab_type.subset[sigtab_type.subset$diff.depth>1,],digits=c(3,3,3,3,3,3,3,6,3,3)), sanitize.text.function = function(x) x)

print(type="html",xtable(sigtab_type.subset[sigtab_type.subset$diff.depth < -1,],digits=c(3,3,3,3,3,3,3,6,3,3)), sanitize.text.function = function(x) x)
```

The OTUs shown are those with mean difference in abundance between the two depths exceeding 1 % of the community. These are a subset of `r nrow(sigtab_type.subset)` OTUs which are shown in full along with adjusted P values in Table S3. 

## Table. Fungi of Kalahari arenosols in southern Botswana that are significantly (adjusted p <0.05) more abundant in grass interspaces with cyanobacterial biocrusts, or in soils lacking cyanobacteria under shrubs and trees. This table is removed from the revised manuscript, replaced by a volcano plot. However, it is kept here for reference.

```{r results='asis',echo=FALSE}
# display an html table of significant OTUs identified by DESEQ

print(type="html",xtable(sigtab_cover.subset[sigtab_cover.subset$diff.cover>1,],digits=c(3,3,3,3,3,3,3,6,3,3)), sanitize.text.function = function(x) x)

print(type="html",xtable(sigtab_cover.subset[sigtab_cover.subset$diff.cover< -1,],digits=c(3,3,3,3,3,3,3,6,3,3)), sanitize.text.function = function(x) x)
```

The OTUs shown are those with mean difference in abundance between canopy (no cyanobacteria) and open (cyanobacterial biocrust present) areas exceeding 1 % of the community. These are a subset of `r nrow(sigtab_cover.subset)` OTUs which differed significantly, all of which are shown in full along with adjusted P values in Table S3. 

## Table X. Fungi of Kalahari arenosols in southern Botswana that are significantly increased in open areas but not significantly correlated with biocrust in open areas.
Not including this table in the manuscript, but leaving here for reference. Note that the top OTUs in this table can be identified in Table 3 (those having no depth association). This info can also be found in supplement bigtable.csv.

```{r results='asis',echo=FALSE}

# identify OTUs that are significantly increased in open areas but not significantly correlated with biocrust in open areas:

sig_otus.open <- row.names(sigtab_cover.subset[sigtab_cover.subset$diff>0,])
sig_otus.crust <- row.names(sigtab_type.subset[order(sigtab_type.subset$diff,decreasing = T),])
biocrust_profile_otus <- sig_otus.open[!sig_otus.open%in%sig_otus.crust]


wanted_cols_profile <- c("Phylum" ,	"Class" ,	"Order" ,	"Family"   , "canopy","open",	"diff.cover",	"padj.cover" ,"crust","subsoil" ,	"diff.depth" ,	"padj.depth"	)

bigtab_cover.subset.profile <- bigtab[biocrust_profile_otus,]
#row.names(bigtab_cover.subset.profile) <- bigtab_cover.subset.profile$Row.names
# order the table by abundance difference between open and canopy zones
bigtab_cover.subset.profile <- bigtab_cover.subset.profile[order(bigtab_cover.subset.profile$diff.cover,decreasing = T),]
# print the top 6
print(type="html",xtable(bigtab_cover.subset.profile[1:6,wanted_cols_profile]), sanitize.text.function = function(x) x)

```


## Mantel comparison of fungal and bacterial communities
Compare community distance matrix of fungi with the bacteria distance matrix using data from Elliott et al (2014). Two tests are run:

- open zones (grass interspace)
- canopy zones (under tree and shrub)

```{r mantel, echo=FALSE}

# get the bacteria OTU table in the same order as the fungi one
otu_table(bacteria) <- otu_table(bacteria)[,sample_names(expt)]

# prepare subset objects for canopy and open areas
fun.open <- subset_samples(expt, cover=="open")
bac.open <- subset_samples(bacteria, cover=="open")
fun.canopy <- subset_samples(expt, cover=="canopy")
bac.canopy <- subset_samples(bacteria, cover=="canopy")

# make distance matrices for open areas
bac.dist.open = phyloseq::distance(bac.open, "bray")
fun.dist.open = phyloseq::distance(fun.open, "bray")

# make distance matrices for canopy areas
bac.dist.canopy = phyloseq::distance(bac.canopy, "bray")
fun.dist.canopy = phyloseq::distance(fun.canopy, "bray")

# Check that the distance matrices are all in the same order
# identical(labels(fun.dist.open),labels(bac.dist.open))
# identical(labels(fun.dist.canopy),labels(bac.dist.canopy))
# yes they are

# run the Mantel tests
mantel(bac.dist.open,fun.dist.open)
mantel(bac.dist.canopy,fun.dist.canopy)

```

# Supplementary Tables & Figures

## Rarefaction
```{r rare,echo=FALSE,fig.width=9, fig.height=6}
exptv <- veganotu(expt)
exptv.individual <- veganotu(expt)
# identical(row.names(exptv),row.names(sample_data(expt)))
row.names(exptv) <- sample_data(expt)$zc_depth 
cols <- as.numeric(as.factor(row.names(exptv)))
rarecurve(exptv,step = 100,label = T,cex=0.6,bty="n",col=cols)
```

## Table S1. Sample data and sample level results
```{r}
rich <- estimate_richness(expt,split=TRUE)
map.df <- cbind(map.df,rich[row.names(map.df),])

write.csv(map.df,file="tables/sample_data.csv")
```
Various tables are in the [tables](tables/) folder. Table S3 is called sample_data.csv.

## Table S2. Relative abundance of fungal phlya in Kalahari arenosols of southern Botswana. 
Overall results are shown in addition to breakdown by depth and nearby vegetation type. 

```{r manuscript_calcs, results='asis',echo=FALSE, warning=FALSE}
a <- as.data.frame(t(otu_table(expt.rel.phylum.all)))
b <- as.data.frame(t(otu_table(expt.rel.phylum.type)))
c <- as.data.frame(t(otu_table(expt.rel.phylum.veg)))

phylum_summary <- cbind(a,b,c)
# order by overall
phylum_summary <- phylum_summary[order(phylum_summary$overall,decreasing = T),] 

print(type="html",xtable(phylum_summary))

```


## Table S3. OTU summary including taxonomy, relative abundance, and statistical results indicating differential abundance by depth and nearby vegetation.

Various tables are in the [tables](tables/) folder. Table S3 is called bigtab.csv.


## Note S1. ADONIS test for community differences with respect to vegetation zone, depth, and sampling month.
This analysis was performed using adonis() function of the vegan package, which is described as follows:
> Analysis of variance using distance matrices - for partitioning distance matrices among sources of variation and fitting linear models (e.g., factors, polynomial regression) to distance matrices; uses a permutation test with pseudo-F ratios.

ADONIS test was done using the whole experiment data at all taxonomic ranks.

```{r adonis,message=FALSE,echo=FALSE,warning=FALSE,cache=TRUE}
my_rank = rank_names(expt)[2]
for (my_rank in c(rank_names(expt)[2:5],"OTU")) {
  cat("\n\n__",my_rank,"__\n")
  if(my_rank!="OTU") {
    expt.taxrank <- tax_glom(physeq=expt,taxrank=my_rank)
  } else {
    expt.taxrank <- expt
  }  
  dist = phyloseq::distance(expt.taxrank, "bray")
  # dist = phyloseq::distance(expt, "unifrac") # Unifrac gives similar results
  NMDS = ordinate(expt.taxrank, "NMDS", dist)
  print(adonis2(dist ~ zone + depth + month , as(sample_data(expt.taxrank), "data.frame")) )
}
```



# Extra supporting information not for publication directly (may inform manuscript or support other result)

## Figure xx. OTU richness estimation (Chao1) and diversity index (Shannon) by site and depth in Kalahari arenosols, southern Botswana. 
The mean and standard error are shown (n=6). Sample coding: AG=annual grass, PG=perennial grass, S=shrub, T=tree.

(not using this now in manuscript, just kept here for reference.)

```{r rich, echo=FALSE, warning=FALSE, fig.width=8, fig.height=4,message=FALSE, results='hide'}
p <- plot_richness(expt, x = "zone_code", color = "depth2", shape = "depth2", measures = c("Chao1","Shannon")) 
p2 <- p + stat_summary(fun.data = "mean_se", position = position_dodge(width = 0.2))

p2$layers <- p2$layers[c(-1,-2)]
p2 <- p2 + theme(aspect.ratio=1) + xlab("vegetation")
(p2 <- p2 + scale_colour_discrete(name="Depth") + scale_shape_discrete(name="Depth"))

p2d <- p2$data
p <- plot_richness(expt, x = "zone_code", color = "depth2", shape = "depth2") 

```

## Table xx. Significance of vegetation type, depth, and season on fungal phylum abundance in Kalahari arenosols of southern Botswana. (open areas only)
wilcox is used for factors with =2 levels (depth, month, and zone).

(not using this now in manuscript, just kept here for reference.)

```{r wilcox_open, echo=FALSE, warning=FALSE, results='asis'}
expt.rel.phylum.open <- subset_samples(expt.rel.phylum, cover=="open")
# remove zero observations
keep <- taxa_sums(expt.rel.phylum.open)>0
expt.rel.phylum.open <- prune_taxa(keep,expt.rel.phylum.open)
# construct a matrix to hoold results
mat <- matrix(nrow=ntaxa(expt.rel.phylum.open),ncol=6, dimnames = list(taxa_names(expt.rel.phylum.open),c("zone.W", "zone.p", "depth.W","depth.p","month.W","month.p")))
for(i in 1:ntaxa(expt.rel.phylum.open)) {
  phylum <- as.numeric(otu_table(expt.rel.phylum.open)[i,])
  phylum_name <- taxa_names(expt.rel.phylum.open)[i]
  depth <- get_variable(physeq = expt.rel.phylum.open, varName = "depth")
  zone <- get_variable(physeq = expt.rel.phylum.open, varName = "zone")
  month <- get_variable(physeq = expt.rel.phylum.open, varName = "month")
  # perform the tests
  d <- wilcox.test(phylum ~ depth) 
  z <- wilcox.test(phylum ~ zone) 
  m <- wilcox.test(phylum ~ month) 
  # store the results
  mat[phylum_name,"depth.W"] <- d$statistic
  mat[phylum_name,"depth.p"] <- d$p.value
  mat[phylum_name,"month.W"] <- m$statistic
  mat[phylum_name,"month.p"] <- m$p.value
  mat[phylum_name,"zone.W"] <- z$statistic
  mat[phylum_name,"zone.p"] <- z$p.value
}

row.names(mat) <- tax_table(expt.rel.phylum)[row.names(mat),2]
print(type="html",xtable(mat,digits=c(0,0,6,0,6,0,6)))
```

## Table xx. Significance of vegetation type, depth, and season on fungal phylum abundance in Kalahari arenosols of southern Botswana.
kruskal is used for factors with >2 levels (zone). wilcox is used for factors with =2 levels (depth and month).

(not using this now in manuscript, just kept here for reference.)

```{r kruskalwilcox, echo=FALSE, warning=FALSE, results='asis'}
# construct a matrix to hold results
mat <- matrix(nrow=ntaxa(expt.rel.phylum),ncol=6, dimnames = list(taxa_names(expt.rel.phylum),c("zone.chisq", "zone.p", "depth.W","depth.p","month.W","month.p")))
for(i in 1:ntaxa(expt.rel.phylum)) {
  phylum <- as.numeric(otu_table(expt.rel.phylum)[i,])
  phylum_name <- taxa_names(expt.rel.phylum)[i]
  depth <- get_variable(physeq = expt.rel.phylum, varName = "depth")
  zone <- get_variable(physeq = expt.rel.phylum, varName = "zone")
  month <- get_variable(physeq = expt.rel.phylum, varName = "month")
  # perform the tests
  d <- wilcox.test(phylum ~ depth) 
  z <- kruskal.test(phylum ~ zone) 
  m <- wilcox.test(phylum ~ month) 
  # store the results
  mat[phylum_name,"depth.W"] <- d$statistic
  mat[phylum_name,"depth.p"] <- d$p.value
  mat[phylum_name,"month.W"] <- m$statistic
  mat[phylum_name,"month.p"] <- m$p.value
  mat[phylum_name,"zone.chisq"] <- z$statistic
  mat[phylum_name,"zone.p"] <- z$p.value
}

row.names(mat) <- tax_table(expt.rel.phylum)[row.names(mat),2]
print(type="html",xtable(mat,digits=c(0,3,6,0,6,0,6)))
```


## Richness / diversity statistical tests

Check effect of depth on richness/diversity:
```{r richness_comparison,echo=FALSE,message=FALSE,warning=FALSE}
# richness category comparison
rich <- estimate_richness(expt, split=TRUE)

# join metadata table to richness results
rich.plus <- cbind(rich[as.character(map.df$X.SampleID),],map.df)

wilcox.test(Shannon ~ type, data=rich.plus)
kruskal.test(Shannon ~ zone_code, data=rich.plus)

wilcox.test(Chao1 ~ type, data=rich.plus, exact=F)
kruskal.test(Chao1 ~ zone_code, data=rich.plus)

# Shannon and Chao1 differ by zone - check further:
library("pgirmess")
cat("Post-hoc test - Shannon diversity")
kruskalmc(Shannon ~ zone,data=rich.plus)
cat("Post-hoc test - Chao1 richness")
kruskalmc(Chao1 ~ zone,data=rich.plus)

# compare diversity by depth in the whole data and separately in each zone
wilcox.test(Shannon ~ type, data=rich.plus)
wilcox.test(Shannon ~ type, data=rich.plus, subset = rich.plus$zone_code=="AG")
wilcox.test(Shannon ~ type, data=rich.plus, subset = rich.plus$zone_code=="PG")
wilcox.test(Shannon ~ type, data=rich.plus, subset = rich.plus$zone_code=="S")
wilcox.test(Shannon ~ type, data=rich.plus, subset = rich.plus$zone_code=="T")

# compare richness by depth in the whole data and separately in each zone
wilcox.test(Chao1 ~ type, data=rich.plus,exact = F)
wilcox.test(Chao1 ~ type, data=rich.plus, subset = rich.plus$zone_code=="AG")
wilcox.test(Chao1 ~ type, data=rich.plus, subset = rich.plus$zone_code=="PG", exact=F)
wilcox.test(Chao1 ~ type, data=rich.plus, subset = rich.plus$zone_code=="S")
wilcox.test(Chao1 ~ type, data=rich.plus, subset = rich.plus$zone_code=="T")


# check for correlation between plant succession and richness/diversity
rich.plus$succession <- 0
rich.plus$succession[rich.plus$zone_code=="AG"] <- 1
rich.plus$succession[rich.plus$zone_code=="PG"] <- 2
rich.plus$succession[rich.plus$zone_code=="S"] <- 3
rich.plus$succession[rich.plus$zone_code=="T"] <- 4

cor.test(x=rich.plus$succession,y=rich.plus$Chao1, method="spearman", exact=F)
cor.test(x=rich.plus$succession,y=rich.plus$Shannon, method="spearman", exact=F)
```



```{r sigplots,echo=FALSE, warning=FALSE, eval=FALSE}
# Save a plot of all the significant OTUs (EVAL=FALSE to save processing time. No need to regenerate the plots every time)

for(otu in row.names(sigtab_type.subset) ) {
    this_otu <- cbind( sample_data(expt.rel.open)[,c("zone","type")] , t(otu_table(expt.rel.open)[otu,]) )
    png(paste("figures_extra/sig_type_", otu,".png", sep=""),units="mm", res=600 , width = 174, height = 174)
    print(ggplot(this_otu,aes_string(x="zone",y=otu,colour="type")) + geom_boxplot() + geom_point() + scale_y_log10()) 
    dev.off()
}
```

```{r sigplots2,echo=FALSE, warning=FALSE, eval=FALSE}
# Save a plot of all the significant OTUs (EVAL=FALSE to save processing time. No need to regenerate the plots every time)

for(otu in row.names(sigtab_cover.subset) ) {
    this_otu <- cbind( sample_data(expt.rel)[,c("cover","type")] , t(otu_table(expt.rel)[otu,]) )
    png(paste("figures_extra/sig_cover_", otu,".png", sep=""),units="mm", res=600 , width = 174, height = 174)
    print(ggplot(this_otu,aes_string(x="type",y=otu,colour="cover")) + geom_boxplot() + geom_point() + scale_y_log10()) 
    dev.off()
}
```





## Caclulations for manuscript text

```{r manuscript_calcs2,echo=FALSE,warning=FALSE}
cat("Individual abundance of ten most abundant taxonomic classes" )
taxa_sums(expt.rel.class.topN)/nsamples(expt.rel.class.topN)
cat("Combined abundance of ten most abundant taxonomic classes" )
sum(taxa_sums(expt.rel.class.topN)/nsamples(expt.rel.class.topN))
cat("number of OTUs significantly more abundant in subsoil (open areas with cyanobacterial crust only):\n")
sum(sigtab_type.subset$diff.depth<0)
cat("number of OTUs significantly more abundant in crust (open areas with cyanobacterial crust only):\n")
sum(sigtab_type.subset$diff.depth>0)
cat("number of OTUs significantly more abundant in open area:\n")
sum(sigtab_cover.subset$diff.cover>0)
cat("number of OTUs significantly more abundant in canopy area:\n")
sum(sigtab_cover.subset$diff.cover<0)
cat("number of OTUs significantly more abundant in open ares but not significantly differing by depth in open areas")
length(biocrust_profile_otus)
cat("number of fungal sequence reads")
summary(sample_sums(expt))
cat("total number of fungal sequence reads")
sum(sample_sums(expt))

```


```{r fig.width=12,fig.height=10,echo=FALSE, message=FALSE, warning=FALSE}
# output some tables...

write.csv(sigtab_cover,file="tables/sigtab_cover.csv")
write.csv(sigtab_type,file="tables/sigtab_type.csv")
write.csv(bigtab,file="tables/bigtab.csv")
write.csv(bigtab_cover.subset.profile,file="tables/bigtab_profile.csv")


rel_otu <- cbind(tax_table(expt.rel),otu_table(expt.rel))
expt.zcdepth <- merge_samples(expt.rel,group = "zc_depth")
expt.zcdepth <- transform_sample_counts(expt.zcdepth, function(x) x/6)
rel_otu_by_factor <- cbind(tax_table(expt.zcdepth),t(otu_table(expt.zcdepth)))

write.csv(rel_otu,file="tables/otus_tax_and_rel_abundance.csv")
write.csv(rel_otu_by_factor,file="tables/otus_tax_and_rel_abundance_by_factor.csv")

# Tables for BES workshop
write.csv(otu_table(expt),"tables/17_otu_count.csv")
write.csv(otu_table(expt.rel)/100,"tables/17_otu.csv")
write.csv(tax_table(expt),"tables/17_tax.csv")

```

```{r pdfout,echo=FALSE,message=FALSE,results='hide'}
# output pdf figures

fig3 <- zone_plot2
# fig4 <- p2 + theme_bw(base_size = 13, base_family = "") + theme(aspect.ratio=1) + theme(legend.position="bottom")
fig4 <- p.ord + theme_bw(base_size = 13, base_family = "") + theme(aspect.ratio=1) 

# individually
pdf(file = "figures/Figure 3.pdf", width=8, height=5)
fig3
dev.off()

#pdf(file = "figures/Figure 4.pdf", width=8, height=6)
#fig4
#dev.off()

pdf(file = "figures/Figure 4.pdf", width=5, height=5)
fig4
dev.off()





```

